### ISSUE TITLE
[RTM] Interop\ICastable\Castable\Castable.cmd CONTRACT VIOLATION by ReadyToRunInfo::TryLookupTypeTokenFromName

### ISSUE THREAD

**Issue by RussKeldorph** [dotnet/coreclr#5241]
Platform: Windows x64
**set COMPlus_GCStress=3**
```
BEGIN EXECUTION
 "C:\git\coreclr\bin\tests\Windows_NT.x64.Checked\Tests\Core_Root\corerun.exe" Castable.exe

Assert failure(PID 8020 [0x00001f54], Thread: 39492 [0x9a44]): CONTRACT VIOLATION by ReadyToRunInfo::TryLookupTypeTokenFromName at "c:\git\coreclr\src\vm\readytoruninfo.cpp" @ 93

THROWS called in a NOTHROW region.

                        CONTRACT in ReadyToRunInfo::TryLookupTypeTokenFromName at "c:\git\coreclr\src\vm\readytoruninfo.cpp" @ 93
VIOLATED-->  CONTRACT in ClassLoader::GetClassValue at "c:\git\coreclr\src\vm\clsload.cpp" @ 859
                        CONTRACT in ClassLoader::FindClassModuleThrowing at "c:\git\coreclr\src\vm\clsload.cpp" @ 1739
                        CONTRACT in ClassLoader::LoadTypeHandleThrowing at "c:\git\coreclr\src\vm\clsload.cpp" @ 2010
                        CONTRACT in ClassLoader::LoadTypeHandleThrowIfFailed at "c:\git\coreclr\src\vm\clsload.cpp" @ 553
                        CONTRACT in ClassLoader::LoadTypeDefOrRefThrowing at "c:\git\coreclr\src\vm\clsload.cpp" @ 2956
                        CONTRACT in SigPointer::GetTypeHandleThrowing at "c:\git\coreclr\src\vm\siginfo.cpp" @ 1007
                        CONTRACT in SigPointer::GetTypeHandleThrowing at "c:\git\coreclr\src\vm\siginfo.cpp" @ 1007
                        CONTRACT in SigPointer::PeekElemTypeNormalized at "c:\git\coreclr\src\vm\siginfo.cpp" @ 2398
                        FAULT_FORBID in ScanStackRoots at "c:\git\coreclr\src\vm\gcenv.ee.cpp" @ 480
                        GCX_NOTRIGGER in ScanStackRoots at "c:\git\coreclr\src\vm\gcenv.ee.cpp" @ 480
                        CANNOTTHROW in ScanStackRoots at "c:\git\coreclr\src\vm\gcenv.ee.cpp" @ 480



CORECLR! CONTRACT_ASSERT + 0x484 (0x00007fff`180b4c60)
CORECLR! BaseContract::DoChecks + 0x1FF (0x00007fff`180b7823)
CORECLR! EEContract::DoChecks + 0x86 (0x00007fff`1820c12e)
CORECLR! ReadyToRunInfo::TryLookupTypeTokenFromName + 0x250 (0x00007fff`185bb738)
CORECLR! ClassLoader::GetClassValue + 0xA65 (0x00007fff`1832d5fd)
CORECLR! ClassLoader::FindClassModuleThrowing + 0xAFE (0x00007fff`1832b526)
CORECLR! ClassLoader::LoadTypeHandleThrowing + 0xA09 (0x00007fff`1833dc91)
CORECLR! ClassLoader::LoadTypeHandleThrowIfFailed + 0xB67 (0x00007fff`1833cfeb)
CORECLR! ClassLoader::LoadTypeDefOrRefThrowing + 0xBE2 (0x00007fff`183389f6)
CORECLR! SigPointer::GetTypeHandleThrowing + 0xF32 (0x00007fff`18572052)
    File: c:\git\coreclr\src\vm\readytoruninfo.cpp Line: 93
    Image: C:\git\coreclr\bin\tests\Windows_NT.x64.Checked\Tests\Core_Root\CoreRun.exe

Expected: 100
Actual: 1
END EXECUTION - FAILED
FAILED
```

http://dotnet-ci.cloudapp.net/job/dotnet_coreclr/job/release_1.0.0/job/x64_checked_windows_nt_gcstress0x3/1/testReport/Interop/_ICastable_Castable_Castable_/_ICastable_Castable_Castable_cmd/

**Comment by RussKeldorph** [comment]
@gkhanna79 @JohnChen0 @fadimounir FYI

**Comment by gkhanna79** [comment]
@RussKeldorph Does this repro in master as well or just release? @fadimounir Can you PTAL at this today?

**Comment by RussKeldorph** [comment]
@gkhanna79 I don't know.  I kicked off build 12 at http://dotnet-ci.cloudapp.net/job/dotnet_coreclr/job/master/job/x64_checked_windows_nt_gcstress0x3/ in case it will help answer that.

**Comment by fadimounir** [comment]
Fixed with #5296
